examples/platforms/samr21/radio.c

```diff
@@ -0,0 +1,611 @@
+    }
+}
+
+static void handleTx(void)
+{
+    if (sTxDone)
+    {
+        sTxDone = false;
+
+#if OPENTHREAD_ENABLE_DIAG
+
+        if (otPlatDiagModeGet())
+        {
+            otPlatDiagRadioTransmitDone(sInstance, &sTransmitFrame, sTxStatus);
+        }
+        else
+#endif
+        {
+            otLogDebgPlat(sInstance, "Radio transmit done, status: %d",
+                          sTxStatus);
+
+            otPlatRadioTxDone(sInstance, &sTransmitFrame, NULL, sTxStatus);
+        }
+    }
+}
+
+/*******************************************************************************
+ * PHY
+ ******************************************************************************/
+
+void PHY_DataInd(PHY_DataInd_t *ind)
+{
+    sReceiveFrame.mPsdu = ind->data;
+    sReceiveFrame.mLength = ind->size + IEEE802154_FCS_SIZE;
+    sReceiveFrame.mPower = ind->rssi;
+
+    sRxDone = true;
+}
+
+void PHY_DataConf(uint8_t status)
+{
+    sTxStatus = OT_ERROR_ABORT;
+
+    if (status == PHY_STATUS_SUCCESS)
+    {
+        sTxStatus = OT_ERROR_NONE;
+    }
+    else if (status == PHY_STATUS_CHANNEL_ACCESS_FAILURE)
+    {
+        sTxStatus = OT_ERROR_CHANNEL_ACCESS_FAILURE;
+    }
+    else if (status == PHY_STATUS_NO_ACK)
+    {
+        sTxStatus = OT_ERROR_NO_ACK;
+    }
+
+    sTxDone = true;
+}
+
+/*******************************************************************************
+ * Platform
+ ******************************************************************************/
+void samr21RadioInit(void)
+{
+    sTransmitFrame.mLength = 0;
+    sTransmitFrame.mPsdu   = sTransmitPsdu;
+
+    sReceiveFrame.mLength = 0;
+    sReceiveFrame.mPsdu = NULL;
+
+    PHY_Init();
+}
+
+void samr21RadioProcess(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    PHY_TaskHandler();
+
+    handleEnergyScan();
+    handleRx();
+    handleTx();
+}
+
+uint32_t samr21RadioRandomGet(void)
+{
+    uint32_t result;
+
+    radioWakeup();
+
+    result = PHY_RandomReq() << 16 | PHY_RandomReq();
+
+    radioRestore();
+
+    return result;
+}
+
+void samr21RadioRandomGetTrue(uint8_t *aOutput, uint16_t aOutputLength)
+{
+    radioWakeup();
+
+    for (uint16_t i = 0; i < aOutputLength / sizeof(uint16_t); i++)
+    {
+        *((uint16_t *)aOutput) = PHY_RandomReq();
+        aOutput += sizeof(uint16_t);
+    }
+
+    for (uint16_t i = 0; i < aOutputLength % sizeof(uint16_t); i++)
+    {
+        aOutput[i] = PHY_RandomReq();
+    }
+
+    radioRestore();
+}
+
+/*******************************************************************************
+ * Radio
+ ******************************************************************************/
+otRadioState otPlatRadioGetState(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    return sState;
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    (void)aInstance;
+
+#ifdef CONF_IEEE_ADDRESS
+    *((uint64_t *)aIeeeEui64) = CONF_IEEE_ADDRESS;
+#else
+    memcpy(aIeeeEui64, sUserRow->mMacAddress, OT_EXT_ADDRESS_SIZE);
```

Comment:
Where is `sUserRow->mMacAddress` set?  It seams to be coming from a static memory address with no explanation as to where that comes from. (`#define SAMR21_USER_ROW 0x804008`)